control/controlclient/client.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package controlclient implements the client for the Tailscale
+// control plane.
+//
+// It handles authentication, port picking, and collects the local
+// network configuration.
+package controlclient
+
+import (
+	"context"
+	"encoding/json"
+	"reflect"
+
+	"tailscale.com/tailcfg"
+	"tailscale.com/types/empty"
+	"tailscale.com/types/netmap"
+	"tailscale.com/types/persist"
+	"tailscale.com/types/structs"
+)
+
+type Status struct {
```

Comment:
can movement happen in its own commits? if anywhere, I'd move this to status.go. But can we just revert the movement for now?

Reply:
okay, split it into separate commits.

I was aiming to have the Status and Client definitions in the same file, so the entire public API is visible at a glance in one place. But it's pretty reasonable with Status in its own file too, and more obvious where to find it when you're looking, so I'll go with that.